.github/workflows/security.yml

```diff
@@ -26,17 +26,79 @@ jobs:
         with:
           ref: main
 
+      - name: Create Trivy cache directory
+        run: mkdir -p /home/runner/vuln-cache
+
       - name: Run Trivy vulnerability scanner in repo mode
         uses: aquasecurity/trivy-action@master
         with:
           scan-type: "fs"
           format: "sarif"
           output: "trivy-lxd-repo-scan-results.sarif"
+          cache-dir: "/home/runner/vuln-cache"
           severity: "LOW,MEDIUM,HIGH,CRITICAL"
 
+      - name: Cache trivy and vulnerability database
+        uses: actions/cache/save@v4
+        with:
+          path: /home/runner/vuln-cache
+          key: trivy-cache
+
       - name: Upload Trivy scan results to GitHub Security tab
         uses: github/codeql-action/upload-sarif@v3
         with:
           sarif_file: "trivy-lxd-repo-scan-results.sarif"
           sha: ${{ github.sha }}
           ref: refs/heads/main
+
+  trivy-snap:
+    name: Trivy vulnerability scanner - Snap
+    runs-on: ubuntu-22.04
+    needs: trivy-repo
+    strategy:
+      matrix:
+        version:
+          - "latest"
+          - "5.21"
+          - "5.0"
+          - "4.0"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```

Comment:
why do we need to checkout the source code of the repository if we are scanning the snap?

Reply:
To get the correct `github.sha` when uploading the alerts to the security tab and upload the results to the correct branch. See [this](https://github.com/canonical/lxd/pull/14145/commits/5f9f0cf88ac7e96ffe02ff08a291e0d707fbdbb0#diff-6102d4f2cddb56c446459bde9bf8e8044b87ce3f98a24c2bba99debfd62461dcR103).

Reply:
I see thanks!

Reply:
Do we need to submit a sha, as its not really accurate at all to take the head of the branch and relate that to a scan of the stable snap, they could be months different.